File: Longest substring without repeating/answer.php
<?php

function longestSubString($str){
    $length = strlen($str);
    
    if($length <= 1)
        return $length;

    $seenCharacters = [];
    $l = $r = $longest = 0;
    
    while($l < $length && $r < $length ){
        $current_char = $str[$r];
        if(array_key_exists($current_char , $seenCharacters)){
            $l = $seenCharacters[$current_char] + 1;
        }

        $seenCharacters[$current_char] = $r;
        $longest = max($longest , $r - $l + 1);
        $r++;
    }
    return $longest;
}
// Here You Can Change The output To See Diffrent Results
echo(longestSubString('abaacdrr'));

File: Longest substring without repeating/answer.py
def longest_substring(s: str) -> int:
    length = len(s)
    
    if length <= 1:
        return length

    seen_characters = {}
    left = longest = 0
    
    for right in range(length):
        current_char = s[right]
        
        if current_char in seen_characters:
            left = max(left, seen_characters[current_char] + 1)

        seen_characters[current_char] = right
        longest = max(longest, right - left + 1)
    
    return longest

# Here You Can Change The output To See Diffrent Results
print(longest_substring('abbcddee'))